src/node_os.cc

```diff
@@ -146,6 +146,40 @@ static void GetTotalMemory(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(amount);
 }
 
+// The calculations in this method are described in the specification:
+// https://w3c.github.io/device-memory/.
+static void GetApproximatedDeviceMemory(
+    const FunctionCallbackInfo<Value>& args) {
+  float approximated_device_memory_gb_ = 0.0;
+  double physical_memory_mb_ =
+      static_cast<double>(uv_get_total_memory() / (1024 * 1024));
+  DCHECK_GT(physical_memory_mb_, 0);
+  int power = 0;
+  int lower_bound = physical_memory_mb_;
+
+  // Extract the most-significant-bit and its location.
+  while (lower_bound > 1) {
+    lower_bound >>= 1;
+    power++;
+  }
+  // The remaining should always be equal to exactly 1.
+  DCHECK_EQ(lower_bound, 1);
+
+  int64_t upper_bound = lower_bound + 1;
+  lower_bound = lower_bound << power;
+  upper_bound = upper_bound << power;
+
+  // Find the closest bound, and convert it to GB.
+  if (physical_memory_mb_ - lower_bound <= upper_bound - physical_memory_mb_)
+    approximated_device_memory_gb_ = static_cast<float>(lower_bound) / 1024.0;
+  else
+    approximated_device_memory_gb_ = static_cast<float>(upper_bound) / 1024.0;
+
+  // Max-limit the reported value to 8GB to reduce fingerprintability of
+  // high-spec machines.
+  if (approximated_device_memory_gb_ > 8) approximated_device_memory_gb_ = 8.0;
```

Comment:
I don't think finger-printability is an issue on Node, but I'm not sure about this, so raising this comment just to get some feedback.

At least on spec, is just a recommendation: https://www.w3.org/TR/device-memory/#computing-device-memory-value

Reply:
Yeah, fingerprinting is not a concern in Node.js. Also, even if it was, why not use `os.totalmem()/1024**3` to get the value?

Reply:
I just ensured that the code is in sync with chrome and according to the documentation.

Reply:
```suggestion
```

I prefer we remove this check